rustls/src/anchors.rs

```diff
@@ -102,6 +102,9 @@ impl RootCertStore {
     }
 
     /// Add a single DER-encoded certificate to the store.
+    ///
+    /// If you are processing a number of root certificates that may include ancient,
+    /// or syntactically invalid certificates, prefer [`RootCertStore::add_parsable_certificates`].
```

Comment:
I think most users aren't equipped to evaluate whether their root store contains ancient or syntactically invalid certificates. Indeed I suspect most people would assume that on an up-to-date system it contains neither!

Instead let's say: This is suitable if you're bringing in a small set of root certificates that you know will parse successfully. If you're parsing a larger collection of certificates, for instance from a system root store, it's better to call [`RootCertStore::add_parsable_certificates`].

Reply:
> I think most users aren't equipped to evaluate whether their root store contains ancient or syntactically invalid certificates.

You're totally right. Thanks for calling that out. I pushed a revised version closer to your wording.

> Indeed I suspect most people would assume that on an up-to-date system it contains neither!

Haha. True, not everyone is a PKI cynic (yet) :zipper_mouth_face: 

>  If you're parsing a larger collection of certificates, for instance from a system root store, it's better to call [RootCertStore::add_parsable_certificates].

We should also adjust the [usage example](https://docs.rs/rustls-native-certs/latest/rustls_native_certs/) from `rustls-native-certs`, as it contradicts this advice and runs the root store certs through an `add` loop. I can look at fixing that next (but might not get to it tonight).

Reply:
Put up a draft for `rustls-native-certs`: https://github.com/rustls/rustls-native-certs/pull/62